prime-router/src/main/kotlin/azure/DatabaseSubmissionsAccess.kt

```diff
@@ -58,41 +61,91 @@ class DatabaseSubmissionsAccess(private val db: DatabaseAccess = DatabaseAccess(
         sendingOrg: String,
         order: SubmissionAccess.SortOrder,
         sortColumn: SubmissionAccess.SortColumn,
-        resultsAfterDate: OffsetDateTime?,
+        cursor: OffsetDateTime?,
+        toEnd: OffsetDateTime?,
         limit: Int,
         klass: Class<T>
     ): List<T> {
-        val column = when (sortColumn) {
-            /* Here is where we can set a column based on sortColumn's enum
-            * value */
-            SubmissionAccess.SortColumn.CREATED_AT -> ACTION.CREATED_AT
-        }
-
-        val sorted = if (order == SubmissionAccess.SortOrder.ASC) {
-            column.asc()
-        } else column.desc()
-
+        val sortedColumn = createColumnSort(sortColumn, order)
+        val condition = createWhereCondition(sendingOrg, cursor, toEnd)
         return db.transactReturning { txn ->
             val query = DSL.using(txn)
                 .selectFrom(ACTION)
-                .where(ACTION.ACTION_NAME.eq(TaskAction.receive).and(ACTION.SENDING_ORG.eq(sendingOrg)))
-                .orderBy(sorted)
-            if (resultsAfterDate != null) {
-                query.seek(resultsAfterDate)
+                .where(condition)
+                .orderBy(sortedColumn)
+
+            /* This nullifies our selection if we're using the 'between'
+            *  where() statement returned from createWhereCondition()
+            *  For this, it's set to run only when a cursor is given
+            *  with no endCursor. */
+            if (cursor != null && toEnd == null) {
+                query.seek(cursor)
             }
             query.limit(limit)
                 .fetchInto(klass)
         }
     }
 
+    /**
+     * @param order sort the table in ASC or DESC order.
+     * @param sortColumn sort the table by specific column; default created_at.
+     * @return a jooq Condition statement to use in where().
+     */
+    private fun createColumnSort(
+        sortColumn: SubmissionAccess.SortColumn,
+        order: SubmissionAccess.SortOrder
+    ): SortField<OffsetDateTime> {
+        val column = when (sortColumn) {
+            /* Decides sort column by enum */
+            SubmissionAccess.SortColumn.CREATED_AT -> ACTION.CREATED_AT
+        }
+
+        val sortedColumn = when (order) {
+            /* Applies sort order by enum */
+            SubmissionAccess.SortOrder.ASC -> column.asc()
+            SubmissionAccess.SortOrder.DESC -> column.desc()
+        }
+
+        return sortedColumn
+    }
+
+    /**
+     * @param sendingOrg is the Organization Name returned from the Okta JWT Claim.
+     * @param cursor is the OffsetDateTime of the last result in the previous list.
+     * @param toEnd is the OffsetDateTime that dictates how far back returned results date.
+     * @return a jooq Condition statement to use in where().
+     */
+    private fun createWhereCondition(
+        sendingOrg: String,
+        cursor: OffsetDateTime? = null,
+        toEnd: OffsetDateTime? = null
+    ): Condition {
```

Comment:
Whenever we add a where condition matching, we should sanity check the indexes.
The current indexes on the Actions table are:
- `action_id ASC`
- `action_name ASC` + `sending_org ASC`
- `created_at ASC` + `sending_org ASC`

Would the database know to use both .. but the else below would match all 3:
`action_name` + `sending_org` + `created_at`

@cwinters-usds Do you know off the top of your head how the query planner works here?